wagtail/documents/models.py

```diff
@@ -53,6 +54,21 @@ class AbstractDocument(CollectionMember, index.Indexed, models.Model):
         index.FilterField('uploaded_by_user'),
     ]
 
+    def clean(self):
```

Comment:
I think this could be worth making a `clean_file` method rather than just `clean`, as it only validates a single field. Also, I think it would be worth adding a call to `full_clean` to the `save` method, to catch programmatically created models too.

These are just minor points though!

Reply:
So I after adding `full_clean` to `save` and changing the name of `clean` function to `clean_file` seems to be breaking. The function `clean_file` isn't being called during the full clean and I am not able to figure out why it's behaving that way :((

Also, save with full_clean seems to be breaking multiple testcases most of them giving the following error `django.core.exceptions.ValidationError: {'file': ['This field cannot be blank.']}`

Test results : 

```
FAILED (failures=1, errors=76, skipped=389, expected failures=14)
```

Reply:
Ah, that's on me! I used the form syntax instead of the model - apologies.

Looks like that's just because our existing test cases don't provide a file. (We have a function for it `get_test_document_file` in tests/utils.py, but we only use it in one place). Since that's a lot of tests to rewrite, happy to leave that out for now.